prereise/gather/demanddata/transportation_electrification/mileage.py

```diff
@@ -0,0 +1,201 @@
+
+    :param list month_days: a list of integers where each value in the list represents
+        the number of days in that month (0 indexed).
+    :param list month_first_day: a list of integers where each value in the list
+        represents the day of the week of the first day of that month (0 indexed).
+    :return: (*np.ndarray*) -- first list stores the day of the week for each day of
+        the year, second list stores the month the day is in for each day of the year.
+
+    """
+    # initalize day counter
+    day_count = 0
+
+    # initialize
+    input_day = np.zeros(365, int)
+
+    # iterate over 12 months
+    for i in range(len(month_days)):
+        # initialize first day of each month for use in the next for loop
+        k = month_first_day[i]
+
+        # Iterate over the day of each month
+        for j in range(month_days[i]):
+
+            # if k < 6: weekday, if k >= 6: weekend
+            if k < 6:
+                input_day[day_count] = 2
+            elif k >= 6:
+                input_day[day_count] = 1
+
+            # restarts the week counter independent of the intial day
+            if k == 7:
+                k = 1
+            else:
+                k += 1
+
+            # increment day counter
+            day_count += 1
+
+    return input_day
+
+
+def get_input_month(month_days: list[int], month_first_day: list[int]) -> np.ndarray:
+    """Determine month of each day
+
+    :param list month_days: a list of integers where each value in the list represents
+        the number of days in that month (0 indexed).
+    :param list month_first_day: a list of integers where each value in the list
+        represents the day of the week of the first day of that month (0 indexed).
+    :return: (*np.ndarray*) -- stores the month the day is in for each day of the year.
+    """
+    # initalize day counter
+    day_count = 0
+
+    # initialize integers because all zeros
+    input_month = np.zeros(365, int)
+
+    # iterate over 12 months
+    for i in range(len(month_days)):
+        # Iterate over the day of each month
+        for j in range(month_days[i]):
+            # Populate input_month with month of each day ex: input_month[100] gives us month number of day 101
+            input_month[day_count] = i
+
+            # increment day counter
+            day_count += 1
+
+    return input_month
+
+
+def get_month2(data: np.array(list[float or int])) -> np.array(int):
+    """Get month2 value from data.
+
+    :param np.array data: the data to get months from.
+    :return: (*np.array*) -- list of months.
+    """
+    return np.array([x[5] % 100 for x in data], int)
+
+
+def get_dayofweek2(data: np.array(list[float or int])) -> np.array(int):
+    """Get dayofweek2 value from data.
+
+    :param np.array data: the data to get day of week from.
+    :return: (*np.array*) -- the day of the week for each entry in data.
+    """
+    return np.array([x[6] for x in data], int)

```

Comment:
Same comment as above: we can set the column type when we instantiate the dataframe, and we can refer to the column by name instead of position to access it.